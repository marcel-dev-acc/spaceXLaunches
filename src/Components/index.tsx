export { default as Background } from "./Background/Background";
export { default as FilterButton } from "./FilterButton/FilterButton";
export { default as LaunchItem } from './LaunchItem/LaunchItem';
export { default as LoadingModal } from './LoadingModal/LoadingModal';
export { default as Logo } from './Logo/Logo';
export { default as ReloadButton } from './ReloadButton/ReloadButton';
export { default as SortButton } from './SortButton/SortButton';