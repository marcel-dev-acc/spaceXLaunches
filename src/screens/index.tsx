export {default as HomeScreen} from './Home/Home';
export {default as LaunchesScreen} from './Launches/Launches';
