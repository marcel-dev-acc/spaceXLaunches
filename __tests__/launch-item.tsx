import 'react-native';
import React from 'react';
import { LaunchItem } from '../src/components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('render launch item', () => {
  const launch = {
    flight_number: 1,
    mission_name: "Some mission name",
    launch_date_unix: 1670854617,
    rocket: {
      rocket_id: "1",
      rocket_name: "Some rocket name",
    },
  };
  const tree: any = renderer.create(<LaunchItem
    launch={launch}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
