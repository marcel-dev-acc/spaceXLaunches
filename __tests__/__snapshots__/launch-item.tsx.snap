// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render launch item 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "margin": 10,
          "padding": 5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 2,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 22,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 28,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(75, 75, 75, 1)",
                      },
                      Object {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                #1
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(75, 75, 75, 1)",
                      },
                      Object {
                        "alignSelf": "center",
                        "marginLeft": 15,
                      },
                    ],
                  ]
                }
              >
                Some mission name
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 11,
                      "fontWeight": "500",
                      "letterSpacing": 0.5,
                      "lineHeight": 16,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(75, 75, 75, 1)",
                      },
                      Object {
                        "alignSelf": "flex-end",
                      },
                    ],
                  ]
                }
              >
                12th Dec 2022
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(75, 75, 75, 1)",
                      },
                      Object {
                        "alignSelf": "flex-end",
                        "marginTop": 5,
                      },
                    ],
                  ]
                }
              >
                Some rocket name
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
